module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1725,6 +1725,23 @@ public function recoverAction()
         } elseif ($username = $this->params()->fromPost('username')) {
             $user = $userService->getUserByUsername($username);
         }
+        //ILS Driver:
+        //if the user hasn't logged in yet, but is found by the ILS, call function
+        //getPatronFromUsername
+        if (!$user && $this->formWasSubmitted() && !empty($username)) {
```

Comment:
I think this, as opposed to updatePassword, should be split to ILS-specific part. And because the ILS's have different functionality, we can't rely on always being able to fetch a user by username. It could also be email address.
It might also make sense to not even try to retrieve a full user in this situation, but just whatever information is needed when the password is reset. In some cases it could be that the ILS supports password reset by way of a recovery token instead of username, but e.g. for Koha we should just need patron id (borrowernumber).